### PR TITLE
Handle the match option like in the Register message.

### PR DESCRIPTION
**Pull request by ralscha**
Handle getRetained different because options.get could return null.

### CODE REVIEW

**Comment by om26er** [comment]
@ralscha There is a CI failure.

File: autobahn/src/main/java/io/crossbar/autobahn/wamp/messages/Subscribe.java [hunk 1]
```diff
@@ -57,8 +58,21 @@ public static Subscribe parse(List<Object> wmsg) {
...
+        if (options.containsKey("match")) {
+            match = (String) options.get("match");
+            if (!Objects.equals(match, MATCH_EXACT) && !Objects.equals(match, MATCH_PREFIX) &&
+                    !Objects.equals(match, MATCH_WILDCARD)) {
+                throw new ProtocolError("match must be one of exact, prefix or wildcard.");
+            }
+        }
+        
+        boolean getRetained = false;
+        if (options.containsKey("get_retained")) {
```
Comment (om26er) [inline autobahn/src/main/java/io/crossbar/autobahn/wamp/messages/Subscribe.java]: Using options.getOrDefault() would be a cleaner approach here.

Comment (ralscha) [reply autobahn/src/main/java/io/crossbar/autobahn/wamp/messages/Subscribe.java]: You are right. Didn't know about this method. I will change it.

**Comment by ralscha** [comment]
Shouldn't the default value for Publish.excludeMe be true.

**Comment by om26er** [comment]
> Shouldn't the default value for Publish.excludeMe be true.

Yeah, you are right. The relevant part of the spec: http://wamp-proto.org/static/rfc/draft-oberstet-hybi-crossbar-wamp.html#rfc.section.14.4.2.1

**Comment by oberstet** [comment]
LGTM, merged. @ralscha cool, thanks for contributing!

**Pull request merged**